Mathlib/Topology/Algebra/InfiniteSum/ConditionalInt.lean

```diff
@@ -0,0 +1,161 @@
+
+section LeAtTop
+
+variable {G : Type*} [AddCommGroup G] [PartialOrder G] [IsOrderedAddMonoid G] [LocallyFiniteOrder G]
+
+lemma symmetricIcc_le_Conditional :
+    (symmetricIcc G).filter ≤ (conditional G).filter :=
+  Filter.map_mono (tendsto_neg_atTop_atBot.prodMk tendsto_id)
+
+instance : (symmetricIcc G).LeAtTop where
+  le_atTop := le_trans symmetricIcc_le_Conditional (conditional G).le_atTop
+
+variable [NoTopOrder G] [NoBotOrder G]
+
+instance : (symmetricIco G).LeAtTop where
+  le_atTop := by
+    rw [symmetricIco, map_le_iff_le_comap, ← @tendsto_iff_comap]
+    exact tendsto_Ico_neg_atTop_atTop
+
+instance : (symmetricIoc G).LeAtTop where
+  le_atTop := by
+    rw [symmetricIoc, map_le_iff_le_comap, ← @tendsto_iff_comap]
+    exact tendsto_Ioc_neg_atTop_atTop
+
+instance : (symmetricIoo G).LeAtTop where
+  le_atTop := by
+    rw [symmetricIoo, map_le_iff_le_comap, ← @tendsto_iff_comap]
+    exact tendsto_Ioo_neg_atTop_atTop
+
+end LeAtTop
+
+end IntervalFilters
+section Int
+
+variable {α : Type*} {f : ℤ → α} [CommGroup α] [TopologicalSpace α] [ContinuousMul α]
+
+lemma symmetricIcc_eq_map_Icc_nat :
+    (symmetricIcc ℤ).filter = atTop.map (fun N : ℕ ↦ Icc (-(N : ℤ)) N) := by
+  simp [← Nat.map_cast_int_atTop, Function.comp_def]
+
+lemma symmetricIcc_eq_symmetricIoo_int : (symmetricIcc ℤ) = (symmetricIoo ℤ) := by
+  simp only [symmetricIcc, symmetricIoo, mk.injEq]
+  ext s
+  simp only [← Nat.map_cast_int_atTop, Filter.map_map, Filter.mem_map, mem_atTop_sets, ge_iff_le,
+    Set.mem_preimage, comp_apply]
+  refine ⟨fun ⟨a, ha⟩ ↦ ⟨a + 1, fun b hb ↦ ?_⟩, fun ⟨a, ha⟩ ↦ ⟨a - 1, fun b hb ↦ ?_⟩⟩ <;>
+  [ convert ha (b - 1) (by grind) using 1; convert ha (b + 1) (by grind) using 1 ] <;>
+  simpa [Finset.ext_iff] using by grind
+
+@[to_additive]
+lemma HasProd.hasProd_symmetricIco_of_hasProd_symmetricIcc {a : α}
+    (hf : HasProd f a (symmetricIcc ℤ)) (hf2 : Tendsto (fun N : ℕ ↦ (f N)⁻¹) atTop (𝓝 1)) :
+    HasProd f a (symmetricIco ℤ) := by
+  simp only [HasProd, tendsto_map'_iff, symmetricIcc_eq_map_Icc_nat,
+    ← Nat.map_cast_int_atTop, symmetricIco] at *
+  apply tendsto_of_div_tendsto_one _ hf
+  simpa [Pi.div_def, fun N : ℕ ↦ prod_Icc_eq_prod_Ico_mul f (show (-N : ℤ) ≤ N by omega)]
+    using hf2
+
+@[to_additive]
+lemma multipliable_symmetricIco_of_multiplible_symmetricIcc
+    (hf : Multipliable f (symmetricIcc ℤ)) (hf2 : Tendsto (fun N : ℕ ↦ (f N)⁻¹) atTop (𝓝 1)) :
+    Multipliable f (symmetricIco ℤ) :=
+  (hf.hasProd.hasProd_symmetricIco_of_hasProd_symmetricIcc hf2).multipliable
+
+@[to_additive]
+lemma tprod_symmetricIcc_eq_tprod_symmetricIco [T2Space α]
+    (hf : Multipliable f (symmetricIcc ℤ)) (hf2 : Tendsto (fun N : ℕ ↦ (f N)⁻¹) atTop (𝓝 1)) :
+    ∏'[symmetricIco ℤ] b, f b = ∏'[symmetricIcc ℤ] b, f b :=
+  (hf.hasProd.hasProd_symmetricIco_of_hasProd_symmetricIcc hf2).tprod_eq
+
+@[to_additive]
+lemma hasProd_symmetricIcc_iff {α : Type*} [CommMonoid α] [TopologicalSpace α]
```

Comment:
Do you think this one and the next ones would have a version that would work over a more general index than Z, for instance Z^d?